docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -0,0 +1,64 @@
+- Add metrics to your data-source. More information [here](/introduction/backend-plugins#collect-metrics).
+- Add tracing to your data-source. In addition to metrics, tracing allows you to deep dive into your plugin performance. More information [here](/create-a-plugin/extend-a-plugin/add-distributed-tracing-for-backend-plugins). And an example [here](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b/examples/datasource-http-backend/pkg/plugin/datasource.go#L141-L156).
+- Keep cached connections. This is an important optimization. More information [here](/introduction/backend-plugins#caching-and-connection-pooling) and an example [here](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b23645251997088ac7a1707a72d3fd9248/examples/datasource-http-backend/pkg/plugin/datasource.go#L40-L66).
+- Add macro support. Macros are similar to variables but they are tipically evaluated in the backend and it allows to return values based on environment data like the current time selection. These are usually defined using the syntax `$__macroName` (e.g. `$__timeFilter`). See this [example](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b23645251997088ac7a1707a72d3fd9248/examples/datasource-basic/pkg/query/macro.go) to discover how you can implement it.
+- For SQL data-sources, check out the [`sqlutil` package in the SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/data/sqlutil). It includes multiple helpers to work with SQL data-sources for things like data frame conversion, default macros, etc.
+- Do not use the local file system. Different plugins share the same environment and for security reasons, plugins should not rely on local files.
+- Similar to the above, avoid the usage of environment variables. Those are also a security risk and should be avoided. For data-source-specific configuration, use the `jsonData` or `secureJsonData` fields in the plugin.json file. In case some generic configuration is needed for the plugin (shared among data-sources), use the [`plugin` configuration](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#pluginplugin_id).
+- Plugins should not execute arbitrary code in the backend. Again, this is a security risk and should be avoided. If your plugin needs to execute code, provide a list of allowed commands and validate the input before executing it.
+- Logged arguments should not expose sensitive information, such as secrets.
+- In general, any error happening should be logged with level `error`.
+- Using the `info` level is normally not recommended: use the `debug` level instead.
```

Comment:
Need to check Marcus' PR on this too just to make sure we're aligned on our advice.

Reply:
same, which PR? seems to be a a few

Reply:
Same one https://github.com/grafana/plugin-tools/pull/547

Reply:
there is a ["Best practices"](https://github.com/grafana/plugin-tools/pull/547/files#diff-0625ec7b29cd40a4e5940aa569122d446643d559b6090e5810c57d378d1136b8R169) there, we can just link to that :+1: